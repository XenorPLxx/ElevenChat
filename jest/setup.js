import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

global.fetch = () => new Promise(() => {});
global.React = React;
global.renderer = renderer;
global.shallow = new ShallowRenderer();
