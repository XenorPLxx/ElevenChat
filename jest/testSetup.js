// @flow

import React from 'react';
import { JSDOM } from 'jsdom';
// import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import type { ComponentType } from 'react';

// import { store } from '../App';

export const setupJestMount = ({
  props,
  Component,
}: // connected = false,
{
  props: Object,
  Component: ComponentType<any>,
  connected: boolean,
}) => {
  require('react-native-mock-render/mock');
  const jsdom = new JSDOM('');
  global.window = jsdom.window;
  global.document = jsdom.window.document;
  Object.keys(global.window).forEach((property) => {
    if (typeof global[property] === 'undefined') {
      global[property] = global.window[property];
    }
  });
  const component = <Component {...props} />;
  // if (connected) {
  //   component = (
  //     <Provider store={store}>
  //       <Component {...props} />
  //     </Provider>
  //   );
  // }
  const wrapper = mount(component);
  return {
    props,
    wrapper,
  };
};

export const setupJestShallow = ({
  props,
  Component,
  lifecycleExperimental = false,
}: {
  props: Object,
  Component: ComponentType<any>,
  lifecycleExperimental: boolean,
}) => {
  const wrapper = shallow(<Component {...props} />, { lifecycleExperimental });
  return {
    props,
    wrapper,
  };
};
