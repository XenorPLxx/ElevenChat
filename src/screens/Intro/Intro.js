// @flow

import React, { Component } from 'react';
import { Button, View, Text } from 'react-native';
import type { NavigationScreenProp } from 'react-navigation';

type Props = {
  navigation: NavigationScreenProp<null>,
};

// eslint-disable-next-line
class Intro extends Component<Props> {
  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text>Intro Screen</Text>
        <Button title="Go to Home" onPress={() => navigation.navigate('HomeScreen')} />
      </View>
    );
  }
}

export default Intro;
