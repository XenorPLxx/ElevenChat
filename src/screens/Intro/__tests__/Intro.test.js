import Intro from '../Intro';
import { setupJestShallow } from '../../../../jest/testSetup';

describe('<Intro />', () => {
  const props = {};

  describe('snapshots', () => {
    it('with default props', () => {
      const { wrapper } = setupJestShallow({
        props: { ...props },
        Component: Intro,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
