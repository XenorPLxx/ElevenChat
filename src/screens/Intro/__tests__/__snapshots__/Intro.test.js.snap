// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> snapshots with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Intro />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component>
          Intro Screen
        </Component>,
        <Button
          onPress={[Function]}
          title="Go to Home"
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Intro Screen",
        },
        "ref": null,
        "rendered": "Intro Screen",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPress": [Function],
          "title": "Go to Home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component>
            Intro Screen
          </Component>,
          <Button
            onPress={[Function]}
            title="Go to Home"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Intro Screen",
          },
          "ref": null,
          "rendered": "Intro Screen",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Go to Home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "lifecycleExperimental": false,
  },
}
`;
